### PR TITLE
Feature/add validation endpoints 171

### PR DESCRIPTION
**Pull request by Christopher-Sandbach**

### CODE REVIEW

**Comment by andykitch** [comment]
Did you mean to remove the tileserver file?

File: src/API/src/validation/validation.controller.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { FileInterceptor } from '@nestjs/platform-express';
+import { ValidationService } from './validation.service';
+
+@Controller('validation')
+export class ValidationController {
+  constructor(private validationService: ValidationService) {}
+
+  @Post('validateUploadBionomics')
+  @UseInterceptors(FileInterceptor('file'))
+  async validateBionomicsCsv(
```
Comment (andykitch) [inline src/API/src/validation/validation.controller.ts:16]: We probably want these endpoints to be restricted to uploader roles

File: src/API/src/validation/validation.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import * as csvtojson from 'csvtojson';
+import { Validator } from '../validation/types/base.validator';
+
+@Injectable()
+export class ValidationService {
+  constructor(private logger: Logger) {}
+
+  async validateBionomicsCsv(csv: string) {
+    const errorLog = [];
+    const rawArray = await csvtojson({
```
Comment (andykitch) [inline src/API/src/validation/validation.service.ts:11]: I'd stick this bit in the try block too, to catch any errors in parsing

**Comment by Christopher-Sandbach** [comment]
Isn't the tileserver lakes.mbtiles file in the incorrect place? It should be generated having run the scripts instead of being present in the codebase from the start

File: src/API/src/validation/utils/typeValidation.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { isBool, isNumber } from './typeValidation';
+
+describe('isBool', () => {
+  it('returns true if input is string including either true or false', () => {
+    const bool = isBool('true');
+    const bool2 = isBool('false');
```
Comment (andykitch) [inline src/API/src/validation/utils/typeValidation.test.ts:6]: yes and no?

File: src/API/src/validation/utils/typeValidation.test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    expect(bool).toEqual(true);
+    expect(bool2).toEqual(true);
+    expect(number).toEqual(false);
+    expect(string).toEqual(false);
+  });
+});
+describe('isNumber', () => {
+  it('returns true if string includes numbers (floats, ints, scientifc)', () => {
+    const int = isNumber('123');
+    const float = isNumber('123.456');
```
Comment (andykitch) [inline src/API/src/validation/utils/typeValidation.test.ts:18]: How about something like `.5`?

**Review by andykitch — APPROVED** [review]

**Comment by ray-millward-tessella** [comment]
> Isn't the tileserver lakes.mbtiles file in the incorrect place? It should be generated having run the scripts instead of being present in the codebase from the start

Lakes is the only one that should be checked in - the process of calling the Open Street Map API to get about 300MB of lake data is very slow for what is actually a small output file. It's a trade off rather than having to wait for it to be built. This is correct currently.

**Pull request merged**